File: TaskApi/Models/Entities/Task.cs
using TaskStatus = TaskApi.Enums.TaskStatus;
using TaskApi.Enums;

namespace TaskApi.Models.Entities
{
    public class Task
    {
        public int Id { get; set; }
        public string Title { get; set; } = string.Empty;
        public string Description { get; set; } = string.Empty;
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
        public DateTime ModifiedAt { get; set; } = DateTime.UtcNow;
        public DateTime DueDate { get; set; }
        public TaskStatus Status { get; set; } = TaskStatus.Pending;
        public Priority Priority { get; set; } = Priority.Medium;
    }
}

File: task-client/src/types/Task.ts
export type TaskStatus = "Pending" | "In Progress" | "Completed" | "Cancelled";
export type Priority = "Low" | "Medium" | "High";

export interface Task {
    id: number;
    title: string;
    description?: string;
    dueDate: string;
    status: TaskStatus;
    priority: Priority;
}

export interface TaskDetail {
    id: number;
    title: string;
    description: string;
    createdAt: string;
    status: TaskStatus;
    priority: Priority;
}

export interface TaskFilter {
    keyword?: string;
    status?: TaskStatus;
    priority?: Priority;
    startDate?: string;
    endDate?: string;
    page?: number;
    size?: number;
}